src/core/deployer.sol

```diff
@@ -12,187 +11,68 @@
+import "./borrower/deployer.sol";
+import "./lender/deployer.sol";
 
-import { Title } from "tinlake-title/title.sol";
-import { LightSwitch } from "./borrower/lightswitch.sol";
-import { Shelf } from "./borrower/shelf.sol";
-import { Distributor } from "./test/simple/distributor.sol";
-import { Pile } from "./borrower/pile.sol";
-import { Collector } from "./borrower/collect/collector.sol";
-import { Principal } from "./borrower/ceiling/principal.sol";
-import { PushRegistry } from 'tinlake-registry/registry.sol';
-
-contract LenderFabLike {
-    function deploy(address,address,address) public returns (address);
-}
-
-contract LenderLike {
-    function rely(address) public;
-    function file(address) public;
-}
-
-contract WardsLike {
-    function rely(address) public;
-}
-
-contract PileFab {
-    function newPile() public returns (Pile pile) {
-        pile = new Pile();
-        pile.rely(msg.sender);
-        pile.deny(address(this));
-    }
-}
-
-contract TitleFab {
-   function newTitle(string memory name, string memory symbol) public returns (Title title) {
-        title = new Title(name, symbol);
-        title.rely(msg.sender);
-        title.deny(address(this));
-    }
-}
+import "tinlake-auth/auth.sol";
 
-contract LightSwitchFab {
-   function newLightSwitch() public returns (LightSwitch lightswitch) {
-        lightswitch = new LightSwitch();
-        lightswitch.rely(msg.sender);
-        lightswitch.deny(address(this));
-    }
-}
+contract MainDeployer is Auth {
+    BorrowerDeployer borrowerDeployer;
+    LenderDeployer lenderDeployer;
 
-contract ShelfFab {
-   function newShelf(address tkn_, address title_, address debt_, address principal_) public returns (Shelf shelf) {
-        shelf = new Shelf(tkn_, title_, debt_, principal_);
-        shelf.rely(msg.sender);
-        shelf.deny(address(this));
+    constructor () public {
+        wards[msg.sender] = 1;
     }
-}
-
-contract DistributorFab {
-    // note: this is the mock distributor, which will interface with the lender/tranche side of Tinlake, and does not require auth for now.
-    function newDistributor(address shelf_, address token_) public returns (Distributor distributor) {
-        distributor = new Distributor(shelf_, token_);
-        return distributor;
+    function file(bytes32 what, address deployer) public auth {
+        if(what == "borrower") borrowerDeployer = BorrowerDeployer(deployer);
+        else if (what == "lender") lenderDeployer = LenderDeployer(deployer);
+        else revert();
     }
-}
 
-contract CollectorFab {
-    function newCollector(address distributor, address shelf, address pile, address threshold) public returns (Collector collector) {
-        collector = new Collector(distributor, shelf, pile, threshold);
-        collector.rely(msg.sender);
-        collector.deny(address(this));
-    }
-}
+    function wireDepends() public auth {
+        address distributor_ = address(lenderDeployer.distributor());
```

Comment:
maybe we also need to make sure here that the addresses already exist

Reply:
yes. I will add more safety checks. It is only called once so it makes sense